r/R/dplyr-datetime-helpers.R

```diff
@@ -240,25 +214,24 @@ build_formats <- function(orders) {
 #' @noRd
 build_format_from_order <- function(order) {
   char_list <- list(
-    "y" = c("%y", "%Y"),
-    "m" = c("%m", "%B", "%b"),
-    "d" = "%d",
-    "H" = "%H",
-    "M" = "%M",
-    "S" = "%S",
-    "I" = "%I"
+    "%T" = "%H-%M-%S",
+    "%y" = c("%y", "%Y"),
+    "%m" = c("%m", "%B", "%b"),
+    "%b" = c("%m", "%B", "%b")
   )
 
-  split_order <- strsplit(order, split = "")[[1]]
+  formats <- stringr::str_extract_all(order, "%{0,1}(O*[a-zA-Z])")[[1]]
```

Comment:
We currently do not depend on {stringr} - i.e. `stringr` is not listed under `Imports` in DESCRIPTION. I'm not sure it's worth the extra dependency for a single function call, which we could reproduce with base R functions.

Reply:
Indeed `%` is already removed in `build_formats`, removing.
Also switching `str_extract_all` to:
```R
formats <- regmatches(order, gregexpr("(O{0,1}[a-zA-Z])", order))[[1]]
```

I do think we need regex to keep `O{format_letter}` formats intact (otherwise `"Om"` -> `c("O", "m")`).
Lubridate also [does test for the O prefix](https://github.com/tidyverse/lubridate/blob/main/tests/testthat/test-parsers.R#L807).